var/spack/repos/builtin/packages/caliper/package.py

```diff
@@ -79,6 +80,8 @@ def cmake_args(self):
         spec = self.spec
 
         args = [
+            ('-DPYTHON_EXECUTABLE=%s' % 
+                os.path.join(spec['python'].prefix.bin, 'python3')),
```

Comment:
```suggestion
                spec['python'].command.path),
```
This works better when, for example, you have multiple Python installations in the same directory. If you build with `python@3.7.5`, it will search for `python3.7`, then `python3`, then `python` in that order.